src/operation/overlayng/OverlayMixedPoints.cpp

```diff
@@ -29,15 +29,48 @@
 #include <geos/operation/overlayng/OverlayUtil.h>
 #include <geos/util/Assert.h>
 
-
-
 namespace geos {      // geos
 namespace operation { // geos.operation
 namespace overlayng { // geos.operation.overlayng
 
-
 using namespace geos::geom;
 
+/**
+ * @brief Extracts and rounds coordinates from a geometry
+ *
+ */
+class CoordinateExtractingFilter: public geom::CoordinateFilter {
+public:
+    CoordinateExtractingFilter(std::unique_ptr<CoordinateArraySequence>& p_pts, const PrecisionModel* p_pm)
+        : pts(p_pts), pm(p_pm)
+    {}
+
+    /**
+     * Destructor.
+     * Virtual dctor promises appropriate behaviour when someone will
+     * delete a derived-class object via a base-class pointer.
+     * http://www.parashift.com/c++-faq-lite/virtual-functions.html#faq-20.7
+     */
+    ~CoordinateExtractingFilter() override {}
+
+    /**
+     * Performs a filtering operation with or on coord in "read-only" mode.
+     * @param coord The "read-only" Coordinate to which
+     * 				the filter is applied.
+     */
+    void
+    filter_ro(const geom::Coordinate* coord) override
+    {
+        Coordinate p(*coord);
+        pm->makePrecise(p);
+        pts->add(p);
+    }
+
+private:
+    std::unique_ptr<CoordinateArraySequence>& pts;
```

Comment:
Just store a `CoordinateArraySequence&` here.

Reply:
So change the constructor to take a `CoordinateArraySequence&`?  if so, how should the client 
's `std::unique_ptr<CoordinateArraySequence>` be passed?

Reply:
Yes, the constructor can take `CoordinateArraySequence&` and the caller can construct it like `CoordinateExtractingFilter filter(*coords, p_pm)`

Reply:
ah, ok, I see that it can be called with `*seq`.